File: backend/app/schemas/mcp_store.py
from pydantic import BaseModel


class StoreEnv(BaseModel):
    name: str
    description: str | None = None
    required: bool = False
    secret: bool = False


class StoreDoc(BaseModel):
    label: str
    url: str


class StoreServer(BaseModel):
    name: str
    suggested_name: str
    title: str | None = None
    description: str | None = None
    version: str | None = None
    transport: str
    package_kind: str | None = None
    package_id: str | None = None
    command: str | None = None
    args: list[str] = []
    url: str | None = None
    env_required: list[StoreEnv] = []
    source_url: str | None = None
    docs: list[StoreDoc] = []
    install_command: str
    mcp_json: dict


class SearchResponse(BaseModel):
    servers: list[StoreServer]
    next_cursor: str | None = None


class PackageDoc(BaseModel):
    description: str | None = None
    readme: str | None = None
    homepage: str | None = None


class ImportRequest(BaseModel):
    name: str  # nome no registry (identificador)
    name_local: str | None = None  # override do name local (default = suggested_name)


class ImportResult(BaseModel):
    id: str
    name: str

File: frontend/src/api/mcpStore.ts
import { onUnauthorized } from "@/lib/session"

export interface StoreEnv {
  name: string
  description: string | null
  required: boolean
  secret: boolean
}

export interface StoreServer {
  name: string
  suggested_name: string
  title: string | null
  description: string | null
  version: string | null
  transport: string
  package_kind: string | null
  command: string | null
  args: string[]
  url: string | null
  env_required: StoreEnv[]
  source_url: string | null
  install_command: string
  mcp_json: Record<string, unknown>
}

export interface ImportResult {
  id: string
  name: string
}

async function request<T>(path: string, init?: RequestInit): Promise<T> {
  const res = await fetch(`/api/v1${path}`, {
    credentials: "include",
    headers: { "Content-Type": "application/json", ...init?.headers },
    ...init,
  })
  if (!res.ok) {
    if (res.status === 401) onUnauthorized()
    const data = await res.json().catch(() => ({}))
    throw new Error(data?.detail ?? `Erro ${res.status}`)
  }
  return res.json()
}

export const mcpStoreApi = {
  search: (q: string) =>
    request<StoreServer[]>(`/mcp-store/search?q=${encodeURIComponent(q)}`),
  import: (name: string, name_local?: string) =>
    request<ImportResult>("/mcp-store/import", {
      method: "POST",
      body: JSON.stringify({ name, name_local }),
    }),
}
